src/esp/io/JsonEspTypes.h

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#ifndef ESP_IO_JSONESPTYPES_H_
+#define ESP_IO_JSONESPTYPES_H_
+
+/** @file
+ * @brief See JsonAllTypes.h. Don't include this header directly in user code.
+ */
+
+#include "JsonBuiltinTypes.h"
+#include "JsonMagnumTypes.h"
+
+#include "esp/assets/RenderAssetInstanceCreationInfo.h"
+#include "esp/core/esp.h"
+#include "esp/gfx/replay/Keyframe.h"
+
+namespace esp {
+namespace io {
+
+inline JsonGenericValue toJsonValue(const esp::vec3f& vec,
+                                    JsonAllocator& allocator) {
+  JsonGenericValue floatsArray(rapidjson::kArrayType);
+  for (int i = 0; i < vec.size(); i++) {
+    floatsArray.PushBack(vec.data()[i], allocator);
+  }
+  return floatsArray;
+}
+
+inline bool fromJsonValue(const JsonGenericValue& floatsArray,
+                          esp::vec3f& vec) {
+  // TODO: helper for objects that are simply arrays, with error-handling
+  ASSERT(floatsArray.Size() == vec.size());
+  for (int i = 0; i < vec.size(); i++) {
+    vec.data()[i] = floatsArray[i].GetFloat();
+  }
+  return true;
+}
+
+inline JsonGenericValue toJsonValue(const esp::geo::CoordinateFrame& frame,
+                                    JsonAllocator& allocator) {
+  JsonGenericValue obj(rapidjson::kObjectType);
+  addMember(obj, "up", frame.up(), allocator);
+  addMember(obj, "front", frame.front(), allocator);
+  addMember(obj, "origin", frame.origin(), allocator);
+  return obj;
+}
+
+inline bool fromJsonValue(const JsonGenericValue& obj,
+                          esp::geo::CoordinateFrame& frame) {
+  esp::vec3f up;
+  esp::vec3f front;
+  esp::vec3f origin;
+  readMember(obj, "up", up);
+  readMember(obj, "front", front);
+  readMember(obj, "origin", origin);
+  frame = esp::geo::CoordinateFrame(up, front, origin);
+  return true;
```

Comment:
Could this one return an accurate bool code based on the results of each readmember? Would be helpful for some places that I would use this in managed containers. perhaps something like

```
<...>
bool success = true;
success &= readMember(obj, "up", up);
success &= readMember(obj, "front", front);
success &= readMember(obj, "origin", origin);
<...>
return success;
```